src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -329,18 +329,34 @@ internal BoundExpression BindValueAllowArgList(ExpressionSyntax node, Diagnostic
                 return null;
             }
 
+            SymbolKind containingMemberOrLambdaKind = this.ContainingMemberOrLambda.Kind;
+            Debug.Assert(containingMemberOrLambdaKind == SymbolKind.Field || containingMemberOrLambdaKind == SymbolKind.Method);
+            bool isMemberInitializer = containingMemberOrLambdaKind == SymbolKind.Field;
```

Comment:
Perhaps a comment that auto-property initializers have already been lowered into initializers on their backing field?